### PR TITLE
New `create_optimizer` class method

### PR DESCRIPTION
**Pull request by mvacaporale**
This introduces a new `create_optimizer` class method to `SupervisedExperiment`.

To the best I can tell, our `Imagenet` experiment is the only one that uses the `batch_norm_weight_decay` and `bias_weight_decay` options. No other config invokes these . Thus, I've removed them from `SupervisedExperiment` and added them to `LegacyImagenetConfig`.

### CODE REVIEW

**Comment by mvacaporale** [comment]
Note, there is a config in the GSC project that uses `batch_norm_weight_decay`, but this will be changed in a future PR. A discussion with @subutai led us to realize the `default_base` config needed a few edits, but those haven't been made yet.

**Review by mrcslws — CHANGES_REQUESTED** [review]
I think this would be better in its own mixin. The `LegacyImagenetConfig` mixin parses old configs dicts and translates them into the new Dataset format. This batch norm weight decay functionality has nothing to do with legacy configs, and is still potentially useful for future experiments.

The ImagenetExperiment class could auto-include the mixin.

**Comment by mvacaporale** [comment]
Agreed. I'd still like the option of setting weight-decay to zero. I also wanted the ability to configure other optimizer param groups via the config, so I've added a new mixin to help with this: `ConfigureOptimizerParamGroups`.

The `LegacyImagementConfig` now adjusts the config for this mixin, and `ImagenetExperiment` automatically includes it.

**Review by mrcslws — APPROVED** [review]
Looks good. Note that our internal `VDropWeightDecay` mixin was overriding the `should_decay_parameter` method (that's why the method existed and was in the execution order). But that mixin isn't currently being used by any experiments, so I suppose we can get rid of it now that there's a new approach for specifying per-parameter weight decay.
